website/integrations: add Atlantis integration (#22888)
goauthentik/authentik: website/integrations/infrastructure/atlantis/index.md

---
title: Integrate with Atlantis
sidebar_label: Atlantis
support_level: community
---

## What is Atlantis?

> Atlantis is an application for automating Terraform via pull requests.
>

> > -- https://www.runatlantis.io/guide

Reviewer: ```suggestion
> -- https://www.runatlantis.io
```

    - **Choose a Provider type**: select **Proxy Provider** as the provider type.
    - **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
        - Set **Mode** to **Forward auth (single application)**.
        - Set **External host** to `https://atlantis.company`.
        - Under **Advanced protocol settings**, set **Unauthenticated Paths** to the following value:

            ```text
            ^/events$
            ```

        - Under **Authentication settings**, disable **Intercept header authentication**.

    - **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/bindings-overview/) (policy, group, or user) to manage the listing and access to applications on a user's **Application Dashboard** page.

3. Click **Submit** to save the new application and provider.

### Configure proxy outpost

The proxy provider requires an authentik proxy outpost. If you do not already have a proxy outpost, follow the [outpost documentation](/docs/add-secure-apps/outposts/) to create and deploy one.

> Add the Atlantis application to the proxy outpost that should serve it:

Reviewer: ```suggestion
Add the Atlantis application to a proxy outpost that will serve it:
```

1. Log in to authentik as an administrator and open the authentik Admin interface.
2. Navigate to **Applications** > **Outposts**.
3. Click the edit icon for the proxy outpost. This can be the built-in **authentik Embedded Outpost** or another proxy outpost.
4. Under **Available Applications**, select the Atlantis application and move it to **Selected Applications**.
5. Click **Update** to save your changes.

## Atlantis configuration

No SSO configuration is required in Atlantis.

Configure your reverse proxy to use the authentik outpost as the forward auth endpoint for `https://atlantis.company`.

Requests to `/outpost.goauthentik.io` must be routed to the authentik outpost. All other requests, including `/events`, must continue to be routed to Atlantis.

> Configure your Git host webhook URL to `https://atlantis.company/events`. The `/events` endpoint is skipped by authentik so that Git host webhooks can reach Atlantis. Atlantis should still validate those webhook requests with its existing webhook secret or webhook authentication settings.

Reviewer: ```suggestion
Set your Git host webhook URL to `https://atlantis.company/events`. The `/events` endpoint is skipped by authentik so that Git host webhooks can reach Atlantis. Atlantis should still validate those webhook requests with its existing webhook secret or webhook authentication settings.
```
